### ISSUE TITLE
When using unsavedChangesAlerts(), DeleteAction and ForceDeleteAction present a second pop-up after a confirmation modal is accepted.

### ISSUE THREAD

**Issue by RalphMRivera** [laravel-filament/filament#11042]
### Package

filament/filament

### Package Version

v3.2.12

### Laravel Version

10.42.0

### Livewire Version

v3.4.1

### PHP Version

8.2.15

### Problem description

When visiting an edit page and deleting or force deleting a record by confirming the modal without first modifying the record, a pop-up warns that unsaved changes will be lost.

### Expected behavior

When I click the delete or force delete button without making any changes and then confirm the deletion in a modal, I should not see a third warning pop-up like this.

<img width="322" alt="image" src="https://github.com/filamentphp/filament/assets/1270019/1e72d18f-7474-4eb0-816d-1d2d5b469fdd">

Additionally, even if changes were made to the record and the user decides to delete the record, the delete or force delete modal should be enough confirmation without an additional browser pop-up.

### Steps to reproduce

1. Add `->unsavedChangesAlerts()` to the public `panel(Panel $panel): Panel` function.
2. Create a resource and add `DeleteAction` and `ForceDeleteAction` to the edit page, for example:

```php
class EditBook extends EditRecord
{
    protected static string $resource = BookResource::class;

    protected function getHeaderActions(): array
    {
        return [
            ViewAction::make(),
            DeleteAction::make(),
            ForceDeleteAction::make(),
            RestoreAction::make(),
        ];
    }
}
```

Here's an example using the Reproduction Repository:

<img width="1136" alt="image" src="https://github.com/filamentphp/filament/assets/1270019/1cd1c8a2-03dc-47aa-b72c-00e9f12bd89d">

### Reproduction repository

https://github.com/RalphMRivera/filament-issue

**Comment by RalphMRivera** [comment]
@danharrin and @zepfietje - I've updated the reproduction repository connected to this issue to Filament 3.2.15 - The issue remains. Video attached. I am not sure if this comment suffices or if I should open a new issue.


https://github.com/filamentphp/filament/assets/1270019/0a21c1a1-7279-41cf-8d32-a6a21b602eb7

**Comment by danharrin** [comment]
Can you make sure your view cache is clear please? I definitely tested this

**Comment by RalphMRivera** [comment]
Hi @danharrin,

I went nuclear and ran the following on the reproduction repository:

```
composer update;
php artisan filament:upgrade;
npm update;
npm run build;
php artisan clear-compiled;
php artisan cache:clear;
php artisan config:cache;
php artisan debugbar:clear;
php artisan event:clear;
php artisan horizon:clear;
php artisan optimize:clear;
php artisan queue:clear;
php artisan route:clear;
php artisan view:clear;
```
I updated and pushed the reproduction repository. 

I then nuked the browser cache and cookies. I did this on the reproduction repository and my current project (a separate Laravle app) and ended up with the same result. Additional note: I disabled all plug ins in Chrome to ensure there was no weird conflict.


https://github.com/filamentphp/filament/assets/1270019/b531509d-9887-43c4-842f-640db2a1b6ff

**Comment by danharrin** [comment]
Should be resolved by a308d570b82ad1a820eb82b91d924a88d2c83992
